tests/test.sh

```diff
@@ -98,6 +98,20 @@ test_jsn() {
   clean_headers_dump "$FILENAME.headers"
 }
 
+test_metrics() {
+  FILENAME="$TEST_OUT_DIR/$1"
+  URL="$MARTIN_URL/_/metrics"
+
+  echo "Testing $1 from $URL"
+  $CURL --dump-header  "$FILENAME.headers" "$URL" | sed -E 's/^(martin_.*?) [\.0-9]+$/\1 NUMBER/g' > "$FILENAME"
+  clean_headers_dump "$FILENAME.headers"
+  $CURL --dump-header  "$FILENAME.gzip.headers" --compressed "$URL" | sed -E 's/^(martin_.*?) [\.0-9]+$/\1 NUMBER/g' > "$FILENAME.gzip"
+  clean_headers_dump "$FILENAME.gzip.headers"
+  # due to slight timing differences, these might be slightly different
+  sed -iE 's/^content-length: [\.0-9]+$/content-length: NUMBER/g' "$FILENAME.headers"
+  sed -iE 's/^content-length: [\.0-9]+$/content-length: NUMBER/g' "$FILENAME.gzip.headers"
```

Comment:
Fun. `-iE` does not do what I thought it does:

> -i[SUFFIX], --in-place[=SUFFIX]
> edit files in place (makes backup if SUFFIX supplied)


```suggestion
  sed --regexp-extended --in-place 's/^content-length: [\.0-9]+$/content-length: NUMBER/g' "$FILENAME.headers"
  sed --regexp-extended --in-place 's/^content-length: [\.0-9]+$/content-length: NUMBER/g' "$FILENAME.gzip.headers"
```